### PR TITLE
Fix collection and channel not match

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
/kind bug
#25662

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -57,8 +57,8 @@ type Cache interface {
 	GetCollectionID(ctx context.Context, database, collectionName string) (typeutil.UniqueID, error)
 	// GetDatabaseAndCollectionName get collection's name and database by id
 	GetDatabaseAndCollectionName(ctx context.Context, collectionID int64) (string, string, error)
-	// GetCollectionInfo get collection's information by name, such as collection id, schema, and etc.
-	GetCollectionInfo(ctx context.Context, database, collectionName string) (*collectionInfo, error)
+	// GetCollectionInfo get collection's information by name or collection id, such as schema, and etc.
+	GetCollectionInfo(ctx context.Context, database, collectionName string, collectionID int64) (*collectionInfo, error)
```
Comment (jaime0815) [inline internal/proxy/meta_cache.go:61]: we'd better add new method to support get collection by id

File: internal/proxy/meta_cache.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ type Cache interface {
 	GetPartitionInfo(ctx context.Context, database, collectionName string, partitionName string) (*partitionInfo, error)
 	// GetCollectionSchema get collection's schema.
 	GetCollectionSchema(ctx context.Context, database, collectionName string) (*schemapb.CollectionSchema, error)
-	GetShards(ctx context.Context, withCache bool, database, collectionName string) (map[string][]nodeInfo, error)
+	GetShards(ctx context.Context, withCache bool, database, collectionName string, collectionID int64) (map[string][]nodeInfo, error)
```
Comment (jaime0815) [inline internal/proxy/meta_cache.go:70]: we'd better add a new method to support get shards by id

File: internal/proxy/meta_cache.go [hunk 3]
```diff
@@ -304,7 +304,14 @@ func (m *MetaCache) GetCollectionInfo(ctx context.Context, database, collectionN
 	if !ok || !collInfo.isCollectionCached() {
 		tr := timerecord.NewTimeRecorder("UpdateCache")
 		metrics.ProxyCacheStatsCounter.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), method, metrics.CacheMissLabel).Inc()
-		coll, err := m.describeCollection(ctx, database, collectionName, 0)
+		var coll *milvuspb.DescribeCollectionResponse
+		var err error
+		if collectionID != 0 {
+			coll, err = m.describeCollection(ctx, database, "", collectionID)
+		} else {
+			coll, err = m.describeCollection(ctx, database, collectionName, 0)
```
Comment (yah01) [inline internal/proxy/meta_cache.go:312]: Divide this method to two versions, one looks up by collection name, and one looks up by collection ID

File: internal/proxy/meta_cache.go [hunk 4]
```diff
@@ -695,8 +702,9 @@ func (m *MetaCache) UpdateCredential(credInfo *internalpb.CredentialInfo) {
 }
 
 // GetShards update cache if withCache == false
-func (m *MetaCache) GetShards(ctx context.Context, withCache bool, database, collectionName string) (map[string][]nodeInfo, error) {
-	info, err := m.GetCollectionInfo(ctx, database, collectionName)
+// if don't know the collectionID, just need to pass 0 here
+func (m *MetaCache) GetShards(ctx context.Context, withCache bool, database, collectionName string, collectionID int64) (map[string][]nodeInfo, error) {
```
Comment (yah01) [inline internal/proxy/meta_cache.go:706]: same as `GetCollectionInfo`

**Review by yah01 — CHANGES_REQUESTED** [review]

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

File: internal/proxy/meta_cache.go [hunk 5]
```diff
@@ -695,8 +702,12 @@ func (m *MetaCache) UpdateCredential(credInfo *internalpb.CredentialInfo) {
 }
 
 // GetShards update cache if withCache == false
-func (m *MetaCache) GetShards(ctx context.Context, withCache bool, database, collectionName string) (map[string][]nodeInfo, error) {
-	info, err := m.GetCollectionInfo(ctx, database, collectionName)
+func (m *MetaCache) GetShards(ctx context.Context, withCache bool, database, collectionName string, collectionID int64) (map[string][]nodeInfo, error) {
+	log.Ctx(ctx).With(
```
Comment (jiaoew1991) [inline internal/proxy/meta_cache.go:706]: Do you need to write `log := log.Ctx(ctx)` like this? 🤔

**Pull request merged**